scipy/special/_softmax.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import division
+import numpy as np
+from scipy.special import logsumexp
+
+
+def softmax(X, theta=1.0, axis=None):
```

Comment:
Is it possible to make `theta` a weight vector and use broadcasting during the multiplication?

Can the axis be `(-1,)`? Not sure what would be right here.